src/components/Pagination/Pagination.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Pagination.scss';
+import Button from '../Button';
+import PaginationButton from './PaginationButton';
+
+const { block, elem } = bem({
+    name: 'Pagination',
+    classnames: styles,
+    propsToMods: ['align']
+});
+
+const Pagination = props => {
+    const {
+        align,
+        currentPage,
+        maxPageButtons,
+        totalPages,
+        prevLabel,
+        nextLabel,
+        lastLabel,
+        firstLabel,
+        onClick,
+        ...rest
+    } = props;
+
+    const defineRange = () => {
+        const showBefore = Math.floor((maxPageButtons - 1) / 2);
+        const start = Math.max(
+            2,
+            Math.min(currentPage - showBefore, totalPages - maxPageButtons + 1)
+        );
+        const end = Math.min(totalPages, start + maxPageButtons - 2);
+        const range = new Array(end - start + 1).fill().map((_, i) => start + i);
+
+        return range;
+    };
+
+    const isPrevDisabled = currentPage === 1;
+    const isNextDisabled = currentPage === totalPages;
+    const range = defineRange();
+
+    const handleClick = e => {
+        const {
+            dataset: { page = null }
+        } = e.target;
+        const pageNum = +page;
+
+        if (!onClick || !pageNum || pageNum === currentPage) {
+            return false;
+        }
+
+        return onClick(e, pageNum);
+    };
+
+    return (
+        <nav {...rest} {...block(props)} aria-label="pagination">
+            {!!firstLabel && (
+                <Button
+                    {...elem('button', props)}
+                    onClick={handleClick}
+                    disabled={isPrevDisabled}
+                    context="link"
+                    data-page={1}
+                    aria-disabled={isPrevDisabled}
+                >
+                    {`\u00ab ${firstLabel}`}
```

Comment:
Would it make sense to style these arrows, instead of concatenating? This button will always rerender now.

Reply:
Rerender because of what? Btw is there a reason to prefer unicode over html entities (like before - or even the `«` character directly)? This syntax reads more complicated than...

```
<Button>
    &laquo; {prevLabel}
</Button>
```

Reply:
This will work too, and it even honors the spacing.

Reply:
> I don't think the assumption that someone might accidentally remove a whitespace is a strong argument. We have reviews for that.
I saw large refactors more then once, that effected such cases. Reviews missed it because they are a needle in a haystack. The latest example was introducing eslint in the current JF stack. Whitespace deletions are really hard to spot, and since we don't have visual regression tests, I would really like to make the code more resilient in this way